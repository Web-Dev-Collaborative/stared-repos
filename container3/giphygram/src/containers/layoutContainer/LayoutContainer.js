import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Layout } from '../../components/shared/layout/Layout';
import { layoutResize } from '../../actions/layoutActions';

export class LayoutContainer extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ]).isRequired,
    layoutResize: PropTypes.func.isRequired,
  };

  onMediaQueryChange = (screenSize) => {
    const { layoutResize: layoutResizeCallback } = this.props;
    layoutResizeCallback(screenSize);
  };

  render() {
    const { children } = this.props;

    return (
      <Layout
        onMediaQueryChange={this.onMediaQueryChange}
        onMediaQueryInit={this.onMediaQueryChange}
      >
        {children}
      </Layout>
    );
  }
}

const mapDispatchToProps = {
  layoutResize,
};

export default connect(
  null,
  mapDispatchToProps,
)(LayoutContainer);
