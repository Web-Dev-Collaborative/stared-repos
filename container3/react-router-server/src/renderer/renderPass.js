import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
