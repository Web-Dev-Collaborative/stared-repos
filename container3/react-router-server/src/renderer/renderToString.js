import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
