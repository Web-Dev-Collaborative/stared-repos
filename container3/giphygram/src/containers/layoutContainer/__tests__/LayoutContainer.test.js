import React from 'react';
import renderer from 'react-test-renderer';
import { LayoutContainer } from '../LayoutContainer';

jest.mock('../../../components/shared/layout/Layout', () => ({
  Layout: 'Layout',
}));

describe('LayoutContainer', () => {
  it('should be rendered correctly by default with one child', () => {
    const layoutResize = jest.fn();

    const tree = renderer
      .create((
        <LayoutContainer layoutResize={layoutResize}>
          <div>Child</div>
        </LayoutContainer>
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered correctly by default with several children', () => {
    const layoutResize = jest.fn();

    const tree = renderer
      .create((
        <LayoutContainer layoutResize={layoutResize}>
          <div>Child #1</div>
          <div>Child #2</div>
        </LayoutContainer>
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should fire layoutResize callback', () => {
    const layoutResize = jest.fn();

    const layoutContainerComponent = renderer
      .create((
        <LayoutContainer layoutResize={layoutResize}>
          <div>Child #1</div>
          <div>Child #2</div>
        </LayoutContainer>
      )).root;

    expect(layoutResize).not.toHaveBeenCalled();

    const layoutComponent = layoutContainerComponent.findByType('Layout');
    expect(layoutComponent).toBeDefined();

    // Call onResize event.
    layoutComponent.props.onMediaQueryChange('sm');

    expect(layoutResize).toHaveBeenCalledTimes(1);
    expect(layoutResize).toHaveBeenLastCalledWith('sm');
  });
});
