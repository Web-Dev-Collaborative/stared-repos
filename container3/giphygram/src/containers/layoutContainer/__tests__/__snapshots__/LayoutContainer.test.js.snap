// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutContainer should be rendered correctly by default with one child 1`] = `
<Layout
  onMediaQueryChange={[Function]}
  onMediaQueryInit={[Function]}
>
  <div>
    Child
  </div>
</Layout>
`;

exports[`LayoutContainer should be rendered correctly by default with several children 1`] = `
<Layout
  onMediaQueryChange={[Function]}
  onMediaQueryInit={[Function]}
>
  <div>
    Child #1
  </div>
  <div>
    Child #2
  </div>
</Layout>
`;
